src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -0,0 +1,626 @@
+
+            if (sourceDirectoryName == String.Empty)
+            {
+                throw new ArgumentException(SR.Argument_EmptyPath, "sourceDirectoryName");
+            }
+
+            if (destinationDirectoryName == String.Empty)
+            {
+                throw new ArgumentException(SR.Argument_EmptyPath, "destinationDirectoryName");
+            }
+
+            Contract.EndContractBlock();
+
+            EnsureStoreIsValid();
+
+            String sourceDirectoryNameFullPath = GetFullPath(sourceDirectoryName);
+            String destinationDirectoryNameFullPath = GetFullPath(destinationDirectoryName);
+
+            try
+            {
+                Directory.Move(sourceDirectoryNameFullPath, destinationDirectoryNameFullPath);
+            }
+            catch (DirectoryNotFoundException)
+            {
+                throw new DirectoryNotFoundException(String.Format(SR.PathNotFound_Path, sourceDirectoryName));
+            }
+            catch (PathTooLongException)
+            {
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw GetIsolatedStorageException(SR.IsolatedStorage_Operation, e);
+            }
+        }
+
+        /*
+         * Public Static Methods
+         */
+        [System.Security.SecuritySafeCritical] // auto-generated
+        public static IsolatedStorageFile GetUserStoreForApplication()
+        {
+            return GetUserStore();
+        }
+
+        [System.Security.SecurityCritical] // auto-generated
+        internal static IsolatedStorageFile GetUserStore()
+        {
+            IsolatedStorageRoot = FetchOrCreateRoot();
+
+            IsolatedStorageFile isf = new IsolatedStorageFile();
+            isf.m_AppFilesPath = IsolatedStorageRoot;
+            return isf;
+        }
+
+        /*
+         * Private Instance Methods
+         */
+        internal string GetFullPath(string partialPath)
+        {
+            Contract.Assert(partialPath != null, "partialPath should be non null");
+
+            int i;
+
+            // Chop off directory separator characters at the start of the string because they counfuse Path.Combine.
+            for (i = 0; i < partialPath.Length; i++)
+            {
+                if (partialPath[i] != Path.DirectorySeparatorChar && partialPath[i] != Path.AltDirectorySeparatorChar)
+                {
+                    break;
+                }
+            }
+
+            partialPath = partialPath.Substring(i);
+
+            return Path.Combine(m_AppFilesPath, partialPath);
+        }
+
+        /*
+         * Private Static Methods
+         */
+        [System.Security.SecurityCritical] // auto-generated
+        private static void CreatePathPrefixIfNeeded(string path)
+        {
+            string root = Path.GetPathRoot(path);
+
+            Contract.Assert(!String.IsNullOrEmpty(root), "Path.GetPathRoot returned null or empty for: " + path);
```

Comment:
We've standardized on Debug.Assert rather than Contract.Assert across the corefx repo.